internal/manifestwork/manifestwork.go

```diff
@@ -175,3 +194,92 @@ func (mwg *manifestWorkGenerator) getOwnedManifestWorks(ctx context.Context, mcm
 
 	return manifestWorkList, err
 }
+
+func (mwg *manifestWorkGenerator) managedClusterKernelMappings(
+	ctx context.Context,
+	mcm hubv1beta1.ManagedClusterModule,
+	kernelVersions []string) []kmmv1beta1.KernelMapping {
+
+	mod := &kmmv1beta1.Module{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      mcm.Name,
+			Namespace: mcm.Spec.SpokeNamespace,
+		},
+		Spec: mcm.Spec.ModuleSpec,
+	}
+
+	logger := log.FromContext(ctx)
+
+	mappings := make([]kmmv1beta1.KernelMapping, 0, len(kernelVersions))
+
+	for _, kernelVersion := range kernelVersions {
+		kernelVersion := strings.TrimSuffix(kernelVersion, "+")
+
+		kernelVersionLogger := logger.WithValues(
+			"kernel version", kernelVersion,
+		)
+
+		mld, err := mwg.kernelAPI.GetModuleLoaderDataForKernel(mod, kernelVersion)
+		if err != nil {
+			kernelVersionLogger.Info("no suitable container image found; skipping kernel version")
+			continue
+		}
+
+		kernelVersionLogger.V(1).Info("Found a valid mapping",
+			"image", mld.ContainerImage,
+		)
+
+		mapping := mwg.mappingFromModuleLoaderData(mld)
+
+		digest, err := mwg.imageDigestForModuleLoaderData(ctx, mld)
+		if err != nil {
+			kernelVersionLogger.Error(err, "skipping image tag replacement")
+		}
+		if digest != "" {
+			mapping.ContainerImage = strings.Split(mapping.ContainerImage, ":")[0] + "@" + digest
+
+			kernelVersionLogger.V(1).Info("replaced container image tag with digest",
+				"image", mld.ContainerImage,
+				"imageWithDigest", mapping.ContainerImage,
+			)
+		}
+
+		mappings = append(mappings, mapping)
+	}
+
+	return mappings
+}
+
+func (mwg *manifestWorkGenerator) mappingFromModuleLoaderData(mld *api.ModuleLoaderData) kmmv1beta1.KernelMapping {
+	return kmmv1beta1.KernelMapping{
+		ContainerImage:       mld.ContainerImage,
+		Literal:              mld.KernelVersion,
+		InTreeModuleToRemove: mld.InTreeModuleToRemove,
```

Comment:
Don't we need all the other fields such as `RegistryTLS`?

Reply:
Why do we need `RegistryTLS`? I believe it's only used to build or sign.

Reply:
What about `preflight`? Isn't it possible that a customer will run preflight only on the spoke?

Reply:
Not as of today. There is no facility to create a `PreflightValidation` resource on the Spoke.

Reply:
Currently there is no support for `Preflight` on the spokes via the `ManagedClusterModule` CRD. If I'm not mistaken, the user has to create their own `ManifestWork` CR, which includes the desired `Preflight` CR.

Reply:
@mresvanis  I meant, isn't it possible that a user will directly apply a `PreflightValidation` resource directly on the spoke.

@qbarrand Thanks for the answer.